libs/sparkpost-lib/src/main/java/com/sparkpost/transport/RestConnection.java

```diff
@@ -302,12 +308,77 @@ private Response receiveResponse(HttpURLConnection conn, Response response) thro
                     response.getResponseCode());
         }
         return response;
-
     }
 
-    private Response receiveEmptyResponse(HttpURLConnection conn, Response response) throws SparkPostException {
+    // This is used to handle 2xx HTTP responses
+    private Response receiveErrorResponse(HttpURLConnection conn, Response response) throws SparkPostException, IOException {
         response.setRequestId(conn.getHeaderField("X-SparkPost-Request-Id"));
-        return response;
+
+        if (conn.getContentLength() == 0) {
+            throw new SparkPostIllegalServerResponseException(
```

Comment:
SparkPostErrorServerResponseException seems to be more appropriate here. Empty response != Illegal server response.